домашняя работа спринт 15

KozlovskiyD opened the pull request.

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12
домашняя работа спринт14
домашняя работа спринт 15

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12
домашняя работа спринт14
домашняя работа спринт 15.1

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/data/dto/TrackDto.kt:
```diff
@@ -0,0 +1,13 @@
+                val releaseDate: String,
+                val primaryGenreName: String,
+                val country: String,
+                val previewUrl: String
```
Это хорошее решение, объявить модель данных для data-слоя 👍

🍏 Видно, какой в коде отступ большой от края. Поля можно выровнять автоформатированием - Ctrl-Alt-L

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/domain/GetTrackUC.kt:
```diff
@@ -0,0 +1,12 @@
+
+@Suppress("CAST_NEVER_SUCCEEDS")
+class GetTrackUC {
+    fun execute(intent: Intent): Track {
+        return intent.getSerializableExtra("track") as Track
+    }
```
⚠ Тут получается, что эта сущность, расположенная в domain, подтягивает специфичную вещь от Android в виде Intent. Обрати внимание на импорт `import android.content.Intent`. 

Лучше получить трек, при помощи intent.getSerializableExtra("track") as Track, прямо в MediaActivity

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/ui/search/TrackListAdapter.kt:
```diff
@@ -26,6 +30,8 @@ class TrackListAdapter(sharedPrefs: SharedPreferences) :
             if (clickDebounce()) {
                 Intent(it.context, MediaActivity::class.java).apply {
+                    putExtra("activity", true)
+                    putExtra("track", tracks[position])
```
🍏 Привыкай выносить "ключи" и подобные им вещи в константы. Это позволит избежать лишних ошибок, если таких "ключей" будет больше

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/presentation/mediaActivity/MediaActivity.kt:
```diff
@@ -0,0 +1,170 @@
+
+
+
+
```
🍏 Не подумай, что придираюсь, но эти пустые отступы лишние)

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/data/trackRepository/mediaPlayers/MediaPlayers.kt:
```diff
@@ -0,0 +1,49 @@
+const val STATE_RELEASE = 4
+
+
+class MediaPlayers(context: Context) : GetCurrentTime {
```
🍏 Можно просто MediaPlayer, без s)

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/presentation/mediaActivity/GetCoverArtwork.kt:
```diff
@@ -0,0 +1,8 @@
+package com.practicum.playlistmaker3.presentation.mediaActivity
+
+interface GetCoverArtwork {
+
+    fun getCoverArtwork(art: String): String {
+        return art.replaceAfterLast('/', "512x512bb.jpg")
+    }
```
🍏 Решение интересное. Можно, конечно, этот метод определить в самом классе Track, как утилитарный, т.е. вспомогательный. Но и так тоже выглядит неплохо

ilshat-abdulin commented on app/src/main/java/com/practicum/playlistmaker3/presentation/mediaActivity/SimpleDataFormat.kt:
```diff
@@ -0,0 +1,10 @@
+import java.text.SimpleDateFormat
+import java.util.*
+
+interface SimpleDataFormat {
+    fun simpleDateFormat(time: String): String {
+        return SimpleDateFormat("mm:ss", Locale.getDefault()).format(time.toLong())
+    }
```
🍏 Можно такие методы выносить в отдельный файл, например, Utils или DateTimeUtils, и использовать их как статичные

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12
домашняя работа спринт14
домашняя работа спринт 15.1
домашняя работа спринт 15.2
